File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

import java.util.ArrayList;

class Solution {

    public void postOrder(TreeNode root, ArrayList<Integer> result){
        if(root==null){
            return;
        }
        postOrder(root.left, result);
        postOrder(root.right, result);
        result.add(root.val);
        return;
    }

    public List<Integer> postorderTraversal(TreeNode root) {
        ArrayList<Integer> result = new ArrayList<>();
        postOrder(root, result);
        return result;
    }
}

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:

        if not root:
            return []

        result = []

        def postOrder(root: Optional[TreeNode], result: List[int]) -> None:

            if not root:
                return

            postOrder(root.left, result)
            postOrder(root.right, result)
            result.append(root.val)

            return

        postOrder(root, result)

        return result

        
        